tests/posix/common/src/pthread.c

```diff
@@ -992,3 +992,53 @@ ZTEST(posix_apis, test_pthread_attr_large_stacksize)
 	zassert_equal(actual_size, expect_size);
 	zassert_ok(pthread_attr_destroy(&attr));
 }
+
+static void *test_pthread_cancel_fn(void *arg)
+{
+	int oldtype = ~PTHREAD_CANCEL_DEFERRED;
+	int oldstate = ~PTHREAD_CANCEL_ENABLE;
+
+	zassert_ok(pthread_setcancelstate(PTHREAD_CANCEL_DEFERRED, &oldtype));
```

Comment:
Should this be `pthread_setcanceltype()` instead?

Reply:
Yes it should ! Sorry